### DISCUSSION TITLE
Call to undefined function Spatie\ImageOptimizer\Optimizers\escapeshellarg()

### DISCUSSION THREAD

**Discussion by dwgdbima** [spatie/laravel-medialibrary#3365] [Q&A]
<img width="1425" alt="Screenshot 2023-09-02 at 15 39 37" src="https://github.com/spatie/laravel-medialibrary/assets/28435448/3e864b97-301c-4c90-8828-4fa352ec629b">
i got "Call to undefined function Spatie\ImageOptimizer\Optimizers\escapeshellarg()" when i call addMedia function. the weird thing is the function going well in local but show this error in hosting.

**Comment by harrowmykel** [comment]
run "composer install" or "composer update"

**Comment by logudotcom** [comment]
While we do the "composer update"  I get the below error? How do I fix that?

`In ProcessExecutor.php line 451:

  [Error]
  Call to undefined function Composer\Util\escapeshellarg()


Exception trace:
  at phar:///opt/cpanel/composer/bin/composer/src/Composer/Util/ProcessExecutor.php:451
 Composer\Util\ProcessExecutor::escapeArgument() at phar:///opt/cpanel/composer/bin/composer/src/Composer/Util/ProcessExecutor.php:405
 Composer\Util\ProcessExecutor::escape() at phar:///opt/cpanel/composer/bin/composer/src/Composer/EventDispatcher/EventDispatcher.php:430
 Composer\EventDispatcher\EventDispatcher->getPhpExecCommand() at phar:///opt/cpanel/composer/bin/composer/src/Composer/EventDispatcher/EventDispatcher.php:368
 Composer\EventDispatcher\EventDispatcher->doDispatch() at phar:///opt/cpanel/composer/bin/composer/src/Composer/EventDispatcher/EventDispatcher.php:129
 Composer\EventDispatcher\EventDispatcher->dispatchScript() at phar:///opt/cpanel/composer/bin/composer/src/Composer/Autoload/AutoloadGenerator.php:455
 Composer\Autoload\AutoloadGenerator->dump() at phar:///opt/cpanel/composer/bin/composer/src/Composer/Installer.php:362
 Composer\Installer->run() at phar:///opt/cpanel/composer/bin/composer/src/Composer/Command/UpdateCommand.php:247
 Composer\Command\UpdateCommand->execute() at phar:///opt/cpanel/composer/bin/composer/vendor/symfony/console/Command/Command.php:298
 Symfony\Component\Console\Command\Command->run() at phar:///opt/cpanel/composer/bin/composer/vendor/symfony/console/Application.php:1040
 Symfony\Component\Console\Application->doRunCommand() at phar:///opt/cpanel/composer/bin/composer/vendor/symfony/console/Application.php:301
 Symfony\Component\Console\Application->doRun() at phar:///opt/cpanel/composer/bin/composer/src/Composer/Console/Application.php:382
 Composer\Console\Application->doRun() at phar:///opt/cpanel/composer/bin/composer/vendor/symfony/console/Application.php:171
 Symfony\Component\Console\Application->run() at phar:///opt/cpanel/composer/bin/composer/src/Composer/Console/Application.php:145
 Composer\Console\Application->run() at phar:///opt/cpanel/composer/bin/composer/bin/composer:93
 require() at /opt/cpanel/composer/bin/composer:29

update [--with WITH] [--prefer-source] [--prefer-dist] [--prefer-install PREFER-INSTALL] [--dry-run] [--dev] [--no-dev] [--lock] [--no-install] [--no-audit] [--audit-format AUDIT-FORMAT] [--no-autoloader] [--no-suggest] [--no-progress] [-w|--with-dependencies] [-W|--with-all-dependencies] [-v|vv|vvv|--verbose] [-o|--optimize-autoloader] [-a|--classmap-authoritative] [--apcu-autoloader] [--apcu-autoloader-prefix APCU-AUTOLOADER-PREFIX] [--ignore-platform-req IGNORE-PLATFORM-REQ] [--ignore-platform-reqs] [--prefer-stable] [--prefer-lowest] [-i|--interactive] [--root-reqs] [--] [<packages>...]`
